catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/InputTransformerWaiter.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.catalog.transform.InputTransformer;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Waits for configured {@link InputTransformer}s. */
+public class InputTransformerWaiter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(InputTransformerWaiter.class);
+
+  private static final String TRANSFORMER_WAIT_TIMEOUT_PROPERTY =
+      "org.codice.ddf.cdm.transformerWaitTimeoutSeconds";
+
+  private static final long DEFAULT_TRANSFORMER_WAIT_TIMEOUT_SECONDS =
+      TimeUnit.MINUTES.toSeconds(5);
+
+  private static final long DEFAULT_TRANSFORMER_CHECK_PERIOD_MILLIS = TimeUnit.SECONDS.toMillis(20);
+
+  private static final String TRANSFORMER_ID_PROPERTY = "id";
+
+  private final long transformerWaitTimeoutMillis;
+
+  private final long transformerCheckPeriodMillis;
+
+  private final InputTransformerIds inputTransformerIds;
+
+  private final List<ServiceReference<InputTransformer>> inputTransformers;
+
+  /**
+   * Creates a new {@code InputTransformerWaiter} and begins waiting for the list of configured
+   * {@link InputTransformer}s. This constructor blocks.
+   *
+   * @param inputTransformerIds list of {@link InputTransformer} ids to wait for
+   * @param inputTransformers {@link InputTransformer}s in the OSGi service registry
+   */
+  public InputTransformerWaiter(
+      InputTransformerIds inputTransformerIds,
+      List<ServiceReference<InputTransformer>> inputTransformers) {
+    this.inputTransformerIds = inputTransformerIds;
```

Comment:
✏️ How about simply calling the next constructor?

Reply:
Have to call a method to get the system property.